# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1494346392 14400
#      Tue May 09 12:13:12 2017 -0400
# Node ID 1a4b4c074800e765ef8e0b95e617f060f2a4774c
# Parent  d691ed3082379ee74181e2312664664c8fb8d2bc
simplify checking evaluation context for classdef methods

* ov-classdef.cc (is_method_executing): Call tree_evaluator::evaluate
  instead of finding the the value of an lvalue object.

diff --git a/libinterp/octave-value/ov-classdef.cc b/libinterp/octave-value/ov-classdef.cc
--- a/libinterp/octave-value/ov-classdef.cc
+++ b/libinterp/octave-value/ov-classdef.cc
@@ -451,19 +451,17 @@ is_method_executing (const octave_value&
 
           octave::tree_parameter_list *pl = uf->is_classdef_constructor ()
             ? uf->return_list () : uf->parameter_list ();
 
           if (pl && pl->size () > 0)
             {
               octave::tree_decl_elt *elt = pl->front ();
 
-              octave_lvalue ref = elt->lvalue (tw);
-
-              octave_value arg0 = ref.value ();
+              octave_value arg0 = tw->evaluate (elt);
 
               if (arg0.is_defined () && arg0.type_name () == "object")
                 {
                   cdef_object arg0_obj = to_cdef (arg0);
 
                   return obj.is (arg0_obj);
                 }
             }
